exchanges/gemini/gemini_websocket.go

```diff
@@ -165,96 +163,169 @@ func (g *Gemini) WsHandleData() {
+			if err != nil {
+				return err
+			}
+			g.Websocket.DataHandler <- result
+		case "unsubscribe":
+			var result wsUnsubscribeResponse
+			err := json.Unmarshal(respRaw, &result)
+			if err != nil {
+				return err
+			}
+			g.Websocket.DataHandler <- result
+		case "initial":
+			var result WsSubscriptionAcknowledgementResponse
+			err := json.Unmarshal(respRaw, &result)
+			if err != nil {
+				return err
+			}
+			g.Websocket.DataHandler <- result
+		case "heartbeat":
+			var result WsHeartbeatResponse
+			err := json.Unmarshal(respRaw, &result)
+			if err != nil {
+				return err
+			}
+			g.Websocket.DataHandler <- result
+		case "update":
+			if curr.IsEmpty() {
+				return fmt.Errorf("%v - `update` response error. Currency is empty %s",
+					g.Name, respRaw)
+			}
+			var marketUpdate WsMarketUpdateResponse
+			err := json.Unmarshal(respRaw, &marketUpdate)
+			if err != nil {
+				return err
+			}
+			g.wsProcessUpdate(marketUpdate, curr)
+		case "candles_1m_updates",
+			"candles_5m_updates",
+			"candles_15m_updates",
+			"candles_30m_updates",
+			"candles_1h_updates",
+			"candles_6h_updates",
+			"candles_1d_updates":
+			var candle wsCandleResponse
+			err := json.Unmarshal(respRaw, &result)
+			if err != nil {
+				return err
+			}
+			for i := range candle.Changes {
+				g.Websocket.DataHandler <- wshandler.KlineData{
+					Timestamp:  time.Unix(int64(candle.Changes[i][0])*1000, 0),
+					Pair:       curr,
+					AssetType:  asset.Spot,
+					Exchange:   g.Name,
+					Interval:   result["type"].(string),
+					OpenPrice:  candle.Changes[i][1],
+					ClosePrice: candle.Changes[i][4],
+					HighPrice:  candle.Changes[i][2],
+					LowPrice:   candle.Changes[i][3],
+					Volume:     candle.Changes[i][5],
 				}
-				var marketUpdate WsMarketUpdateResponse
-				err := json.Unmarshal(resp.Raw, &marketUpdate)
-				if err != nil {
-					g.Websocket.DataHandler <- err
-					continue
+			}
+
+		default:
+			g.Websocket.DataHandler <- wshandler.UnhandledMessageWarning{Message: g.Name + wshandler.UnhandledMessage + string(respRaw)}
+			return nil
+		}
+	} else if _, ok := result["result"]; ok {
+		switch result["result"].(string) {
+		case "error":
+			if _, ok := result["reason"]; ok {
+				if _, ok := result["message"]; ok {
+					return errors.New(result["reason"].(string) + " - " + result["message"].(string))
 				}
-				g.wsProcessUpdate(marketUpdate, resp.Currency)
-			default:
-				g.Websocket.DataHandler <- fmt.Errorf("%v - unhandled data %s",
-					g.Name, resp.Raw)
 			}
+			return fmt.Errorf("%v Unhandled websocket error %s", g.Name, respRaw)
+		default:
+			g.Websocket.DataHandler <- wshandler.UnhandledMessageWarning{Message: g.Name + wshandler.UnhandledMessage + string(respRaw)}
+			return nil
 		}
 	}
+	return nil
+}
+
+func responseToStatus(response string) order.Status {
+	switch response {
+	case "accepted":
+		return order.New
+	case "booked":
+		return order.Active
+	case "filled":
```

Comment:
https://docs.gemini.com/websocket-api/#filled

Please rename to "fill", we'll need to add a note here that an order is considered "filled" when the remaining_amount is 0